lib/cart/apis/cart_api.dart

```diff
@@ -8,15 +9,39 @@ const PATH = '/occ/v2/electronics-spa/users/anonymous/carts/';
 const PARAMS =
     '?fields=DEFAULT,potentialProductPromotions,appliedProductPromotions,potentialOrderPromotions,appliedOrderPromotions,entries(totalPrice(formattedValue),product(images(FULL),stock(FULL)),basePrice(formattedValue,value),updateable),totalPrice(formattedValue),totalItems,totalPriceWithTax(formattedValue),totalDiscounts(value,formattedValue),subTotal(formattedValue),totalUnitCount,deliveryItemsQuantity,deliveryCost(formattedValue),totalTax(formattedValue,%20value),pickupItemsQuantity,net,appliedVouchers,productDiscounts(formattedValue),user,saveTime,name,description&lang=en&curr=USD';
 
+const NEW_CART_PARAMS =
+    'DEFAULT,potentialProductPromotions,appliedProductPromotions,potentialOrderPromotions,appliedOrderPromotions,entries(totalPrice(formattedValue),product(images(FULL),stock(FULL)),basePrice(formattedValue,value),updateable),totalPrice(formattedValue),totalItems,totalPriceWithTax(formattedValue),totalDiscounts(value,formattedValue),subTotal(formattedValue),totalUnitCount,deliveryItemsQuantity,deliveryCost(formattedValue),totalTax(formattedValue,%20value),pickupItemsQuantity,net,appliedVouchers,productDiscounts(formattedValue),user&lang=en&curr=USD';
+
 class CartApi {
   final http = Dio();
 
   Future<Cart> fetchCart({String? cartCode}) async {
-    final response = await GetIt.instance
-        .get<Dio>(instanceName: Singletons.hybrisClient)
-        .get('${dotenv.get('HYBRIS_BASE_URL')}$PATH$cartCode$PARAMS');
+    try {
+      final response = await GetIt.instance
+          .get<Dio>(instanceName: Singletons.hybrisClient)
+          .get('${dotenv.get('HYBRIS_BASE_URL')}$PATH$cartCode$PARAMS');
+
+      return Cart.fromJson(response.data);
+    } on DioException catch (e) {
+      // try to deserialize error response
+      if (e.response?.data != null) {
```

Comment:
I would be more strict on the condition to know if it's a regular Hybris error.

We may face some situation where we can have a DioException without the expected fields in `data` (server is down or in maintenance, IDK).

I would go with

```
if (e.response?.data?['errors']?.isNotEmpty) {
```

something like that